ddtrace/contrib/multiprocessing/patch.py

```diff
@@ -0,0 +1,105 @@
+from ddtrace.vendor.wrapt import wrap_function_wrapper as _w
+
+import multiprocessing
+
+try:
+    # py3
+    from multiprocessing.process import BaseProcess as Process
+except ImportError:
+    from multiprocessing.process import Process
+
+from ... import Pin, config
+from ...ext import SpanTypes
+from ...internal.logger import get_logger
+from ...utils.formats import asbool, get_env
+from ...utils.importlib import func_name
+from ...utils.wrappers import unwrap as _u
+
+log = get_logger(__name__)
+
+DATADOG_PATCHED = "__datadog_patch"
+DATADOG_PIN = "__dd_pin"
+DATADOG_CONTEXT = "__datadog_context"
+
+# Configure default configuration
+config._add(
+    "multiprocessing",
+    dict(
+        service_name=get_env("multiprocessing", "service_name", "multiprocessing"),
+        app="multiprocessing",
+    ),
+)
+
+
+def patch():
+    if getattr(multiprocessing, DATADOG_PATCHED, False):
+        return
+    setattr(multiprocessing, DATADOG_PATCHED, True)
+
+    pin = Pin(service=config.multiprocessing.service_name, app=config.multiprocessing.app)
+    pin.onto(multiprocessing)
+
+    _w(Process, "__init__", _patch_process_init)
+    _w(Process, "run", _patch_process_run)
+
+
+def unpatch():
+    if getattr(multiprocessing, DATADOG_PATCHED, False):
+        return
+
+    _u(Process, "__init__")
+    _u(Process, "run")
+
+
+def _patch_process_init(wrapped, instance, args, kwargs):
+    pin = Pin.get_from(multiprocessing)
+
+    if not pin or not pin.enabled():
+        return wrapped(*args, **kwargs)
+
+    ctx = pin.tracer.get_call_context().clone()
+    datadog_kwargs = {DATADOG_PIN: pin, DATADOG_CONTEXT: ctx}
+
+    # ``kwargs`` argument to Process can either be the 5th and final positional
+    # argument or a named arguments
+    kwargs_in_args = len(args) == 5
+
+    if kwargs_in_args:
+        init_kwargs = args[-1]
```

Comment:
I'd do `args[4]` instead. If we know that is where it is, if the API changes, then this might break.

Although, if it changes and `kwargs` moves to another spot, same problem, but... I guess /shrug

Reply:
naw, this is probably fine.

Reply:
Since `kwargs_in_args` being true implies `len(args)` being `5`, I guess `4` and `-1` are indeed the same. :)